File: app/src/main/java/com/klinker/android/twitter_l/data/roomdb/entities/Emoji.java
package com.klinker.android.twitter_l.data.roomdb.entities;


import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.Index;
import androidx.room.PrimaryKey;



@Entity(tableName = "emojis", indices = {@Index(value = "icon", unique = true)})
public class Emoji {

    @PrimaryKey(autoGenerate = true)
    public final Long id;

    @ColumnInfo
    public String text;

    @ColumnInfo
    public String icon;

    @ColumnInfo
    public int count;

    //maybe add account???

    public Emoji(long id, String text, String icon, int count) {
        this.id = id;
        this.text = text;
        this.icon = icon;
        this.count = count;
    }

    @Ignore
    public Emoji(String text, String icon) {
        this.id = null;
        this.text = text;
        this.icon = icon;
        this.count = 0;
    }


}

File: app/src/main/java/com/klinker/android/twitter_l/data/roomdb/entities/Emoji.kt
package com.klinker.android.twitter_l.data.roomdb.entities


import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.Index
import androidx.room.PrimaryKey


@Entity(tableName = "emojis",
        indices = [Index(value = ["icon"], unique = true), Index(value = ["text"], unique = true), Index(value = ["count"])])
class Emoji {

    @PrimaryKey(autoGenerate = true)
    val id: Long?


    @ColumnInfo
    var text: String

    @ColumnInfo
    var icon: String

    @ColumnInfo
    var count: Int = 0

    //maybe add account???

    constructor(id: Long, text: String, icon: String, count: Int) {
        this.id = id
        this.text = text
        this.icon = icon
        this.count = count
    }

    @Ignore
    constructor(text: String, icon: String) {
        this.id = null
        this.text = text
        this.icon = icon
        this.count = 0
    }

    fun incrementCount() {
        count += 1
    }

}
